android-core/src/main/java/com/mparticle/modernization/data/MParticleDataRepository.kt

```diff
@@ -0,0 +1,10 @@
+package com.mparticle.modernization.data
+
+import com.mparticle.internal.messages.BaseMPMessage
+
+class MParticleDataRepository {
```

Comment:
Can you explain what the purpose of the Data Repository is?

Reply:
The purpose is to have an abstraction managing data operation and multiple data sources if needed.
For example we migh want to retrieve something from a service (backendApi class), and while retrieving it, parallely save it in to a internal database and/or internal shared preferences to be used kind of a cache layer or so..
The advantange of the repository is just to make one call like "getUser" and hide the details of where the data came from, since all data sources should be injected, and business logic should handle it.